storage/country_tree.hpp

```diff
@@ -18,12 +21,14 @@ class CountryTree
   /// \brief m_children contains all first generation descendants of the node.
   /// Note. Once created the order of elements of |m_children| should not be changed.
   /// See implementation of AddAtDepth and Add methods for details.
-  vector<unique_ptr<CountryTree<T>>> m_children;
+  vector<shared_ptr<CountryTree<T>>> m_children;
```

Comment:
Ok. Let's discuss it. 
I keep pointers to these nodes in CountryTreeFacade::m_countryTreeHashTable for quick access without going throw all the tree.

The other variant to keep unique_ptr here and raw pointers in m_countryTreeHashTable.